unleash-yggdrasil/src/lib.rs

```diff
@@ -28,7 +32,47 @@ pub struct CompiledToggle {
     pub name: String,
     pub enabled: bool,
     pub compiled_strategy: RuleFragment,
-    pub variants: Option<Vec<Variant>>,
+    pub variants: Vec<CompiledVariant>,
+    pub yes: AtomicU32,
+    pub no: AtomicU32,
+    pub default_variant: AtomicU32,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct CompiledVariant {
+    pub name: String,
+    pub weight: i32,
```

Comment:
I'm curious about this type. Can weight be negative and is it only int?

Reply:
So this is Yggdrasil being sandwiched between two competing types. Unleash-types passes weight as an i32 and our hashing library requires a u32. To use weight there needs to be a cast somewhere. I believe the reason we set the weight to an i32 in Unleash-types is because we trust the rigor of our data validation in the typescript version as far as we can throw it and it's better to parse the wrong type and cast it than it is to fail the parse even though you have the data you need